client-internal/src/main/java/org/terracotta/angela/client/Tsa.java

```diff
@@ -374,19 +376,31 @@ public Tsa activateAll() {
 
       if (topology.isNetDisruptionEnabled()) {
         Map<ServerSymbolicName, Integer> proxyTsaPorts = updateToProxiedPorts();
-        int proxyPort = proxyTsaPorts.get(terracottaServer.getServerSymbolicName());
-        IgniteCallable<ConfigToolExecutionResult> callable = () -> {
-          return Agent.controller.configTool(this.instanceId, terracottaServer, cliEnv, "set", "-s", "localhost:" + terracottaServer.getTsaPort(),
-              "-c", "stripe.1.public-hostname=localhost", "-c", "stripe.1.public-port=" + proxyPort);
-        };
-        IgniteClientHelper.executeRemotely(ignite, terracottaServer.getHostname(), ignitePort, callable);
+        for (int i = 0; i < topology.getStripes().size(); ++i) {
+          List<TerracottaServer> terracottaServerList = topology.getStripes().get(i);
+          for (int j = 0; j < terracottaServerList.size(); ++j) {
+            TerracottaServer server = terracottaServerList.get(j);
+            int proxyPort = proxyTsaPorts.get(server.getServerSymbolicName());
+            int stripeId = i + 1;
+            int nodeId = j + 1;
+            String publicHostName = "stripe." + stripeId + ".node." + nodeId + ".public-hostname=" + server.getHostName();
+            String publicPort = "stripe." + stripeId + ".node." + nodeId + ".public-port=" + proxyPort;
+            IgniteCallable<ConfigToolExecutionResult> callable = () -> {
+              return Agent.controller.configTool(this.instanceId, terracottaServer, cliEnv,
+                  "set", "-s", server.getHostName() + ":" + server.getTsaPort(),
+                  "-c", publicHostName, "-c", publicPort);
+            };
+            ConfigToolExecutionResult executionResult = IgniteClientHelper.executeRemotely(ignite, terracottaServer.getHostname(), ignitePort, callable);
+            assertThat(executionResult.getExitStatus(), equalTo(0));
```

Comment:
you don't want to use a hamcrest matcher, it would send an AssertionError, this is not the kind of exception to be sent by the framework, here you should manually send a RuntimeException with the appropriate message